pkg/telemetry/signalanddatastats.go

```diff
@@ -0,0 +1,94 @@
+package telemetry
+
+import (
+	"fmt"
+	"time"
+
+	"go.uber.org/atomic"
+
+	"github.com/livekit/protocol/livekit"
+	"github.com/livekit/protocol/utils"
+)
+
+const statsReportInterval = 10 * time.Second
+
+// stats for signal and data channel
+type SignalAndDataStats struct {
+	trackID         livekit.TrackID
+	pID             livekit.ParticipantID
+	send, recv      atomic.Uint64
+	lastStatsReport atomic.Value // *time.Time
+	telemetry       TelemetryService
+}
+
+func NewBytesTrackStats(datatrack bool, pID livekit.ParticipantID, telemetry TelemetryService) *SignalAndDataStats {
```

Comment:
Had suggested earlier making separate objects. In that case, this becomes a generic `BytesTrackStats` objects and some factory function which creates one of these based on signal or data. That way, this object remains just a track bytes counter with a trackID and does not need to know about signal or data.